cl_headers/module.modulemap

```diff
@@ -0,0 +1,16 @@
+module cl12spir {
```

Comment:
Can we re-use the existing `module.modulemap` from `clang/lib/Headers`?

Reply:
Technically yes, but we have the motivation to split out the common built-ins which doesn't relate to cl12/cl20 version, to reduce the duplication in generated pcm. So it's better we have a customized modulemap in our own repo.